### PR TITLE
feat(ruleset-bundler): a number of minor enhancements

### PR DESCRIPTION
**Pull request by P0lip**
Related to https://github.com/stoplightio/platform-internal/pull/7403

This is an accumulated set of changes I've kept track of after the initial release of the package and the actual integrations attempts in Studio and Spectral CLI.
I suggest reviewing this PR commit by commit because each change is somewhat different.

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/ruleset-bundler/src/plugins/commonjs.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as commonjs } from '@rollup/plugin-commonjs';
```
Comment (marbemac) [inline packages/ruleset-bundler/src/plugins/commonjs.ts:1]: How is consumer expected to import this - deep import through `/dist` or something? (or is not meant for general consumption)

File: packages/ruleset-bundler/src/utils/dedupeRollupPlugins.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// this function makes sure we can only have one plugin with the same name
+// the last plugin definition has a precedence
+import type { Plugin } from 'rollup';
+
+export function dedupeRollupPlugins(plugins: Plugin[]): Plugin[] {
```
Comment (marbemac) [inline packages/ruleset-bundler/src/utils/dedupeRollupPlugins.ts:5]: 👌 very nice

File: packages/ruleset-bundler/src/plugins/commonjs.ts (hunk 1 above)
Comment (P0lip) [reply packages/ruleset-bundler/src/plugins/commonjs.ts:1]: we have package exports, so can can import this as follows
```ts
import { commonjs } from '@stoplight/spectral-ruleset-bundler/plugins/commonjs';
```

**Review by marbemac — DISMISSED** [review]
👏

**Pull request merged**